src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1159,6 +1159,11 @@ object Types {
     private[this] var lastSymbol: Symbol = _
     private[this] var checkedPeriod = Nowhere
 
+    /** Provided for debugging, can be used to intercept assignments of uncommon denotations
+     */  
+    private def checkDenot(d: Denotation)(implicit ctx: Context) = {
```

Comment:
ie rename `lastDenotation` to `_lastDenotation` 
and add `def lastDenotation` and `def lastDenotation_=` methods.